openvdb/unittest/TestAttributeSet.cc

```diff
@@ -598,23 +602,35 @@ TestAttributeSet::testAttributeSet()
 
         CPPUNIT_ASSERT_EQUAL(openvdb::Index(50), attrSet.getConst("float1")->size());
 
-        AttributeSet attrSet2(attrSet, /*arrayLength=*/200);
+        // error as the new length is greater than the data size of the
+        // 'boolNonUniform' attribute
```

Comment:
I'm sure we've talked about this, but a git blame shows that this ValueError has been there since the AttributeArray was first contributed to OpenVDB. What's the reason for this particular check again?

Also, remind me what you're using at DNeg - have you exposed support for dynamic attribute arrays from Houdini? I know in the public repo, it's only available from the API.

Reply:
> What's the reason for this particular check again?

I guess that it just doesn't generally make sense to hold a dynamic array with less values than points (size)? And that you can very easily index into invalid data. Perhaps there were more implementation specific reasons but I can't remember them now.

> Also, remind me what you're using at DNeg

afaik we're not actively using non constant (dynamic) strides

Reply:
The reason I caught this is because of our point merge code which does support this (and it threw up a issue on syncing form the public repo)

Reply:
> I guess that it just doesn't generally make sense to hold a dynamic array with less values than points (size)? And that you can very easily index into invalid data. Perhaps there were more implementation specific reasons but I can't remember them now.

I suppose that's dependent on whether or not you have points with zero values in the dynamic array. I'm not sure it makes sense to disallow that, but this PR probably isn't the right place to change this.

> The reason I caught this is because of our point merge code which does support this (and it threw up a issue on syncing from the public repo)

A point merge unit test? We should really get dynamic attribute array support into Houdini, it's come up a couple of times recently that this is missing.

Reply:
Agreed on both counts